src/ArbiterClientShim.h

```diff
@@ -0,0 +1,90 @@
+/* Copyright (c) 2015-2017 Stanford University
+ *
+ * Permission to use, copy, modify, and distribute this software for any
+ * purpose with or without fee is hereby granted, provided that the above
+ * copyright notice and this permission notice appear in all copies.
+ *
+ * THE SOFTWARE IS PROVIDED "AS IS" AND THE AUTHOR(S) DISCLAIM ALL WARRANTIES
+ * WITH REGARD TO THIS SOFTWARE INCLUDING ALL IMPLIED WARRANTIES OF
+ * MERCHANTABILITY AND FITNESS. IN NO EVENT SHALL AUTHORS BE LIABLE FOR
+ * ANY SPECIAL, DIRECT, INDIRECT, OR CONSEQUENTIAL DAMAGES OR ANY DAMAGES
+ * WHATSOEVER RESULTING FROM LOSS OF USE, DATA OR PROFITS, WHETHER IN AN
+ * ACTION OF CONTRACT, NEGLIGENCE OR OTHER TORTIOUS ACTION, ARISING OUT OF
+ * OR IN CONNECTION WITH THE USE OR PERFORMANCE OF THIS SOFTWARE.
+ */
+
+#ifndef ARBITER_CLIENT_SHIM_H
+#define ARBITER_CLIENT_SHIM_H
+
+#include <stdint.h>
+#include <vector>
+
+#include "Semaphore.h"
+#include "CoreArbiterClient.h"
+
+
+namespace Arachne {
+
+typedef int core_t;
+
+/**
+  * This class functions as a shim for the CoreArbiter client so that Arachne
+  * can run without the Core Arbiter when the arbiter is deactivated.
+  */
+class ArbiterClientShim : public CoreArbiter::CoreArbiterClient {
+  public:
+
+    core_t blockUntilCoreAvailable();
+    bool mustReleaseCore();
+    void setRequestedCores(std::vector<uint32_t> numCores);
+    void unregisterThread();
+    void reset() {
+        currentRequestedCores = 0;
+        currentCores = 0;
+        inactiveCores.reset();
+    }
+
+    static ArbiterClientShim* getInstance() {
```

Comment:
Why have only a single instance? Also, it's good to avoid statics if possible; if other things, they can result in problems with initialization order. Let's discuss.

Also, every method should have a header comment.